packages/spanner/test/session-pool.js

```diff
@@ -1376,23 +1345,143 @@ describe('SessionPool', function() {
+          callback();
+        };
+
+        sessionPool.getNextAvailableSession_(OPTIONS, done);
+      });
+    });
+  });
 
-          assert.strictEqual(event, 'available');
+  describe('pollForSession_', function() {
+    var setInterval_;
 
-          assert.strictEqual(
-            onAvailable(),
-            sessionPool.getNextAvailableSession_(OPTIONS, assert.ifError)
-          );
+    beforeEach(function() {
+      setInterval_ = global.setInterval;
+      global.setInterval = function(cb) {
+        global.setInterval.calledWith_ = arguments;
+        setImmediate(cb);
+        return 1;
+      };
+    });
 
+    afterEach(function() {
+      global.setInterval = setInterval_;
+    });
+
+    it('should capture the acquire request', function() {
+      var callback = function() {};
+
+      sessionPool.pollForSession_(callback);
+      assert.strictEqual(sessionPool.pendingAcquires.length, 1);
+
+      var pendingAcquire = sessionPool.pendingAcquires[0];
+
+      assert.strictEqual(pendingAcquire.callback, callback);
+      assert.strictEqual(pendingAcquire.timeout, sessionPool.acquireTimeout);
+    });
+
+    it('should bail if it detects an acquire interval exists', function() {
+      var callback = function() {};
+
+      sessionPool.acquireInterval = 1;
+      sessionPool.pollForSession_(callback);
+
+      assert.strictEqual(global.setInterval.calledWith_, undefined);
+    });
+
+    it('should poll for a free session', function() {
+      var callback = function() {};
+
+      sessionPool.pollForSession_(callback);
+
+      var intervalArgs = global.setInterval.calledWith_;
+
+      assert.strictEqual(sessionPool.acquireInterval, 1);
+      assert.strictEqual(typeof intervalArgs[0], 'function');
+      assert.strictEqual(intervalArgs[1], 30000);
+    });
+
+    it('should call getNextAvailableSession_ if a read free', function(done) {
+      sessionPool.getNextAvailableSession_ = function(callback) {
+        callback();
+      };
+
+      sessionPool.pool = { free: true };
+      sessionPool.pollForSession_(done);
+    });
+
+    it('should call getNextAvailableSession_ if a write free', function(done) {
+      sessionPool.getNextAvailableSession_ = function(callback) {
+        callback();
+      };
+
+      sessionPool.pool = { free: false };
+      sessionPool.writePool = { free: true };
+      sessionPool.pollForSession_(done);
+    });
+
+    it('should clear the interval when no pending acquires', function(done) {
+      var clearInterval_ = global.clearInterval;
+      var clearCalled = false;
+
+      sessionPool.pool = { free: true };
+
+      sessionPool.getNextAvailableSession_ = function() {
+        sessionPool.pendingAcquires = [];
+      };
+
+      global.clearInterval = function(handle) {
+        assert.strictEqual(handle, sessionPool.acquireInterval);
+
+        setImmediate(function() {
+          assert.strictEqual(sessionPool.acquireInterval, null);
+
+          global.clearInterval = clearInterval_;
```

Comment:
Should pop this up to the first line after `global.clearInterval` is re-assigned. (or just capture the real value in the `before` and re-assign in the beforeEach and afterEach)